packages/flutter_tools/lib/src/resident_devtools_handler.dart

```diff
@@ -71,34 +69,34 @@ class FlutterResidentDevtoolsHandler implements ResidentDevtoolsHandler {
   // This must be guaranteed not to return a Future that fails.
   @override
   Future<void> serveAndAnnounceDevTools({
-    Uri devToolsServerAddress,
-    @required List<FlutterDevice> flutterDevices,
+    Uri? devToolsServerAddress,
+    required List<FlutterDevice?> flutterDevices,
```

Comment:
Is it valid to have nulls in this list (and those in the other methods here)?

Reply:
See https://github.com/flutter/flutter/pull/105798/files/d337ee3f6ded659db096998343f7772f41bf8289#diff-2fb6523413045c1c558c5c124f3ec97d6f307079f129584d81d02ba59486ef0cR174

Reply:
hmm that didn't work

Reply:
look at waitForExtensionForDevice, it returns `null` for the FlutterDevice if it fails

Reply:
good to know. thanks